src/folio/service-points-api.ts

```diff
@@ -1,15 +1,50 @@
 import FolioAPI from "./folio-api.js"
-import { CqlParams, ServicePoint } from '../schema.js'
+import { ServicePoint, ServicePointDetails, CqlParams, Location, Library } from '../schema.js'
 
 interface ServicePointsResponse {
   servicepoints: ServicePoint[]
 }
 
 export default class ServicePointsAPI extends FolioAPI {
+  // This is the maximum also used in type-apis
+  static maxQueryLimit = 2147483647
+  // Removing the individual getServicePoint by id method, since not used anywhere
+  // And will be updated with upcoming pull requests
+  // Parse the description string and add the resulting JSON object in the details filed
+  addDetails(servicePoint: ServicePoint) {
+    return {
+      details: parseServicePointDetails(servicePoint),
+      ...servicePoint
+    }
+  }
+
+  // Get a list of all service points along with the details information required for paging
   async getServicePoints(params: Partial<{ params: CqlParams, [key: string]: object | object[] | undefined }>): Promise<ServicePoint[]> {
+    // Unless the limit is being sent in as a parameter, pass in the max limit
+    // We also want to retain any other parameters being sent through the query
+    if(! ("params" in params && "limit" in params["params"])) {
+      if(! ("params" in params)) {
+        params["params"] = {"limit": ServicePointsAPI.maxQueryLimit}
+      } else {
+        params["params"]["limit"] = ServicePointsAPI.maxQueryLimit
+      }
+    }
```

Comment:
in my experience, this function (or the other versions like it) is only ever called in two ways:
- when you want every single service point
- when you want every single service point that matches a filter

because of this, I don't think we'll ever pass in a `limit` (and we've assumed in other places that we won't want to either). I think it's OK to remove this entirely and just always use the hardcoded max limit.

Reply:
Since limit is still allowed in the CQLParams being sent in here, I am not comfortable with not considering the input unless I also disallow limit as a parameter the user can select: https://github.com/sul-dlss/folio-graphql/blob/service-point-details/src/schema.graphql#L42 .  We could also simply not hardcode the limit at all and just pass in a large number to the query when we ask it. 

The other pattern for locations etc. uses the type api code (https://github.com/sul-dlss/folio-graphql/blob/service-point-details/src/folio/type-api.ts) which does not allow CQL Params to be passed in through the GraphQL query itself.

Reply:
Looking at the pattern, for instance, for items which does allow CQL params in from the UI/query itself, I'm going to go ahead and remove the hard-coded limit. That seems more consistent overall.

Reply:
Now I've gone in the other direction! I removed CQL Params from the schema.graphql file, so the UI and/or input query cannot have limit/query etc. defined as a parameter.  Internally, the code can use these but they won't be exposed in the UI/API.  getServicePoints set the limit to the max number.

Reply:
I think this is ready to go now? @thatbudakguy @cbeer @marlo-longley, hopefully!